sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -0,0 +1,43 @@
+import { Controller, Get, Param } from "@nestjs/common";
+import { StudentRestrictionService } from "../../services";
+import BaseController from "../BaseController";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { UserGroups } from "../../auth/user-groups.enum";
+import { AllowAuthorizedParty, Groups } from "../../auth/decorators";
+import { StudentRestrictionSummary } from "./models/restriction.dto";
+/**
+ * Controller for Restrictions.
+ * This consists of all Rest APIs for restrictions.
+ */
+@Controller("restrictions")
+export class RestrictionController extends BaseController {
+  constructor(
+    private readonly studentRestrictionService: StudentRestrictionService,
+  ) {
+    super();
+  }
+
+  /**
+   * Rest API to get restrictions for a student.
+   * @param studentId
+   * @returns Student Restrictions.
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Get("/student/:studentId")
```

Comment:
are we not using lazy loading

Reply:
No @ann-aot. Considering the number of restrictions a student can have as per discussion with @abschwenker , lazy doesn't seem to benefit on this scenario.

Reply:
Yes you are right @ann-aot , we should use lazy loading subsequently. @dheepak-aot can we change the routes to get take and skip values? it will eventually change the queries. Thanks

Reply:
@ann-aot @guru-aot considering the number of restrictions that a student is going to have as per my discussion with @abschwenker lazy loading is not going to benefit on this scenario.

Reply:
Okay I agree, as the list is going to be less than 20. For this scenario, we can go with complete loading.